File: CodingChallengeSolutionsC++/781. Rabbits in Forest/781.RabbitsinForest.py
class RabbitsinForest:
    def numRabbits(self, answers: List[int]) -> int:
        unique = {}
        rabbits = 0
        for x in answers:
            if x in unique:
                unique[x] = unique[x] + 1
            else:
                unique[x] = 1

        for key, value in unique.items():
            rabbits += math.ceil(value / (key +1)) * (key+1)
        return rabbits
        

File: CodingChallengeSolutionsC++/781. Rabbits in Forest/781.RabbitsinForest.cpp

#include<vector>
#include<map>

class RabbitsinForest {
public:
    int numRabbits(std::vector<int>& answers) {
        std::map<int, int> unique;
        int rabbits = 0;
        for (int x = 0; x < answers.size(); ++x)
        {
            unique[answers[x]]++;
        }

        for (auto& entry : unique)
        {
            rabbits += ceil((double)entry.second / (entry.first + 1)) * (entry.first + 1);
        }
        return rabbits;
    }
};